core/camel-base/src/main/java/org/apache/camel/processor/aggregate/StringAggregationStrategy.java

```diff
@@ -63,12 +63,12 @@ public StringAggregationStrategy pick(Expression expression) {
      */
     @Override
     public Exchange aggregate(Exchange oldExchange, Exchange newExchange) {
-        StringBuffer result; // Aggregate in StringBuffer instead of StringBuilder, to make it thread safe
```

Comment:
StringBuffer is here intentionally because it is mutable and ensures locks. As aggregate method can be called in many threads in parallel, you need to ensure locking in different way (synchronized block, ReentrantLock, ...) before changing this to StringBuilder, otherwise the output will be corrupted.

Reply:
Ok, so maybe it's just simpler to keep StringBuffer... I removed the changes on that file. Thanks